neo4j/v1/session.py

```diff
@@ -256,6 +256,32 @@ def consume(self):
             self.connection = None
         return self._summary
 
+    def single(self):
+        """ Return the next record, failing if none or more than one remain.
+        """
+        records = list(self)
+        num_records = len(records)
+        if num_records == 0:
+            raise ResultError("No records found in stream")
+        elif num_records != 1:
+            raise ResultError("Multiple records found in stream")
```

Comment:
*single* - consume all remaining records and return the first of these, failing if there is not exactly one record